src/uwtools/nice_dict.py

```diff
@@ -0,0 +1,173 @@
+# (C) Copyright 2020-2022 UCAR
+#
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Part of this software is developed by the Joint Center for Satellite Data Assimilation (JCSDA) together with its partners.
+#
+# (C) Copyright 2022-2022 NOAA
+
```

Comment:
Are any of these functions used anywhere in the code base?  Why do we need this external library?  What does Python not do by itself that this does?  How does using this file benefit users/developers versus not using it?